Jenkinsfile

```
@@ -45,36 +45,75 @@ node {
             sh "mv $jfrogCliRepoDir/jfrog builder/"
 
             // Extract CLI version
-            sh 'builder/jfrog --version > version'
-            version = readFile('version').trim().split(" ")[2]
+            version = sh(script: "builder/jfrog -v | tr -d 'jfrog version' | tr -d '\n'", returnStdout: true)
             print "CLI version: $version"
         }
 
-        stage('Download tools cert') {
-            // Download the certificate file and key file, used for signing the JFrog CLI binary.
-            sh """#!/bin/bash
-               builder/jfrog rt dl installation-files/certificates/jfrog/ --url https://entplus.jfrog.io/artifactory --flat --access-token=$DOWNLOAD_SIGNING_CERT_ACCESS_TOKEN
-                """
-
-            sh 'tar xvzf jfrogltd_signingcer_full.tar.gz'
-        }
-
         if ("$EXECUTION_MODE".toString().equals("Publish packages")) {
+            buildRpmAndDeb(version, architectures)
+
             stage('Npm Publish') {
-                print "publishing npm package"
                 publishNpmPackage(jfrogCliRepoDir)
             }
 
             stage('Build and Publish Docker Image') {
                 buildPublishDockerImage(version, jfrogCliRepoDir)
             }
         } else if ("$EXECUTION_MODE".toString().equals("Build CLI")) {
+            downloadToolsCert()
             print "Uploading version $version to Bintray"
             uploadCli(architectures)
         }
     }
 }
 
+def downloadToolsCert() {
+    stage('Download tools cert') {
+        // Download the certificate file and key file, used for signing the JFrog CLI binary.
+        sh """#!/bin/bash
+            builder/jfrog rt dl installation-files/certificates/jfrog/ --url https://entplus.jfrog.io/artifactory --flat --access-token=$DOWNLOAD_SIGNING_CERT_ACCESS_TOKEN
+            """
+
+        sh 'tar xvzf jfrogltd_signingcer_full.tar.gz'
+    }
+}
+
+def buildRpmAndDeb(version, architectures) {
+    boolean built = false
+    for (int i = 0; i < architectures.size(); i++) {
+        def currentBuild = architectures[i]
+        if (currentBuild.debianImage) {
+            stage("Build debian ${currentBuild.pkg}") {
+                build(currentBuild.goos, currentBuild.goarch, currentBuild.pkg, 'jfrog')
+                dir("$jfrogCliRepoDir") {
+                    sh "build/deb_rpm/build-scripts/pack.sh -b jfrog -v $version -f deb --deb-arch $currentBuild.debianArch --deb-build-image $currentBuild.debianImage"
+                    built = true
+                }
+            }
+        }
+        if (currentBuild.rpmImage) {
+            stage("Build rpm ${currentBuild.pkg}") {
+                build(currentBuild.goos, currentBuild.goarch, currentBuild.pkg, 'jfrog')
+                dir("$jfrogCliRepoDir") {
+                    sh "build/deb_rpm/build-scripts/pack.sh -b jfrog -v $version -f rpm --rpm-build-image $currentBuild.rpmImage"
```

Lets run the builtin tests after build:
```suggestion
                    sh "build/deb_rpm/build-scripts/pack.sh -b jfrog -v $version -f rpm --rpm-build-image $currentBuild.rpmImage" -t
```